testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -500,11 +497,18 @@ def test_fasta2select_ClustalW(self, tmpdir):
         # numbers computed from alignment with clustalw 2.1 on Mac OS X
         # [orbeckst] length of the output strings, not residues or anything
         # real...
-        assert len(sel['reference']) == 23080, ("selection string has"
-                                                "unexpected length")
-        assert len(
-            sel['mobile']) == 23090, "selection string has unexpected length"
-
+        assert len(sel['reference']) == 23080, self.error_msg
+        assert len(sel['mobile']) == 23090, self.error_msg
+
+    def test_fasta2select_nogap(self, tmpdir):
+        """test align.fasta2select() on aligned FASTA with no gap in resid
```

Comment:
Also change the function name to something like `test_fasta2select_resids` and the doc string to something like "test align.fasta2select() when resids provided (Issue 3124)"